lib/Doctrine/Common/Cache/PhpFileCache.php

```diff
@@ -101,8 +93,14 @@ protected function doSave($id, $data, $lifeTime = 0)
             'data'      => $data
         );
 
-        $value  = var_export($value, true);
-        $code   = sprintf('<?php return %s;', $value);
+        $serializeValue = serialize($value);
+
+        if (strpos($serializeValue, 'r:') !== false || (is_object($data) && !method_exists($data, '__set_state'))) {
```

Comment:
This code assumes only the root object needs __set_state(), however every child needs the method as well. its very dangerous to mix this logic here with two completely different serialization approaches.